src/ray/object_manager/object_buffer_pool.cc

```diff
@@ -0,0 +1,202 @@
+#include "ray/object_manager/object_buffer_pool.h"
+
+namespace ray {
+
+ObjectBufferPool::ObjectBufferPool(const std::string &store_socket_name,
+                                   const uint64_t chunk_size,
+                                   const int release_delay)
+    :chunk_size_(chunk_size) {
+  store_socket_name_ = store_socket_name;
+  ARROW_CHECK_OK(store_client_.Connect(store_socket_name_.c_str(), "", release_delay));
+}
+
+ObjectBufferPool::~ObjectBufferPool(){
+  // Abort everything in progress.
+  auto get_buf_state_copy = get_buffer_state_;
+  for (const auto &pair : get_buf_state_copy) {
+    AbortGet(pair.first);
+  }
+  auto create_buf_state_copy = create_buffer_state_;
+  for (const auto &pair : create_buf_state_copy) {
+    AbortCreate(pair.first);
+  }
+  RAY_CHECK(get_buffer_state_.empty());
+  RAY_CHECK(create_buffer_state_.empty());
+  ARROW_CHECK_OK(store_client_.Disconnect());
+}
+
+uint64_t ObjectBufferPool::GetNumChunks(uint64_t data_size) {
+  return static_cast<uint64_t>(ceil(static_cast<double>(data_size) / chunk_size_));
+}
+
+uint64_t ObjectBufferPool::GetBufferLength(uint64_t chunk_index, uint64_t data_size){
+  return (chunk_index + 1) * chunk_size_ > data_size
+         ? data_size % chunk_size_
+         : chunk_size_;
+}
+
+std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> ObjectBufferPool::GetChunk(const ObjectID &object_id,
+                                                              uint64_t data_size,
+                                                              uint64_t metadata_size,
```

Comment:
The `metadata_size` argument doesn't appear to be used, other than for the debug statement.